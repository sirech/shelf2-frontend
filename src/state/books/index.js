import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as reducer } from './reducers'
