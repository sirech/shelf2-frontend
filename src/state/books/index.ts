import * as constants from './constants'
import * as actions from './slice'

export { actions, constants }
export { default as booksSelector } from './selectors'
export { default as reducer } from './slice'
