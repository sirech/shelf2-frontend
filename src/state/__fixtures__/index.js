import R from 'ramda'

import books from './books'
import years from './years'
import form from './form'
import login from './login'

const modal = () => ({
  opened: false
})

const search = R.pipe(books, R.omit(['activeYear']))

const state = () => {
  return {
    books: books(),
    search: search(),
    years: years(),
    form: form(),
    modal: modal(),
    login: login()
  }
}

export { books, search, years, form, login }
export default state
