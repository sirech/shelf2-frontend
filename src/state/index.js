import { combineReducers } from 'redux'

import * as R from 'ramda'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as formConstants, reducer as formReducer } from './form'
import { constants as modalConstants, reducer as modalReducer } from './modal'
import { constants as loginConstants, reducer as loginReducer } from './login'

export const actionPicker = list => {
  return actions => {
    const h = R.pick(list)(actions)

    if (Object.keys(h).length !== list.length) {
      throw new Error(
        `Not all requested actions (${list}) exist in the given actions`
      )
    }

    return h
  }
}

const rootReducer = combineReducers({
  [booksConstants.namespace]: booksReducer,
  [searchConstants.namespace]: searchReducer,
  [yearsConstants.namespace]: yearsReducer,
  [formConstants.namespace]: formReducer,
  [modalConstants.namespace]: modalReducer,
  [loginConstants.namespace]: loginReducer,
})

export default rootReducer
