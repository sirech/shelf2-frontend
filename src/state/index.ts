import { ThunkAction, UnknownAction, configureStore } from '@reduxjs/toolkit'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as loginConstants, reducer as loginReducer } from './login'

const store = configureStore({
  reducer: {
    [booksConstants.namespace]: booksReducer,
    [searchConstants.namespace]: searchReducer,
    [yearsConstants.namespace]: yearsReducer,
    [loginConstants.namespace]: loginReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  UnknownAction
>

export default store
