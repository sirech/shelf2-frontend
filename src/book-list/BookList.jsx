// @flow

import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'

import booksSelector from './selectors'

import { actions } from 'state/books'
import Category from 'molecules/category'

type Props = {
  match: { params: { year?: string } },
}

const BookList = ({
  match: {
    params: { year },
  },
}: Props) => {
  const categories = useSelector(booksSelector)
  const dispatch = useDispatch()

  useEffect(() => {
    if (year) {
      dispatch(actions.fetchBooks(year))
    }
  }, [year, dispatch])

  return (
    <div>
      {categories.map(({ name, books }) => (
        <Category key={name} name={name} books={books} />
      ))}
    </div>
  )
}

export default BookList
