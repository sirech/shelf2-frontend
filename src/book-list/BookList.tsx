import React, { useEffect } from 'react'

import { useParams } from 'react-router'
import booksSelector from './selectors'

import { useAppDispatch, useAppSelector } from 'hooks'
import { actions } from 'state/books'
import Category from 'components/category'

const BookList = () => {
  const { year } = useParams<string>()
  const categories = useAppSelector(booksSelector)
  const dispatch = useAppDispatch()

  useEffect(() => {
    if (year) {
      dispatch(actions.fetchBooks(year))
    }
  }, [year, dispatch])

  return (
    <div>
      {categories.map(({ name, books }) => (
        <Category key={name} name={name} books={books} />
      ))}
    </div>
  )
}

export default BookList
