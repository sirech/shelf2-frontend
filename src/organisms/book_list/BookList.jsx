// @flow

import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import * as R from 'ramda'

import booksSelector from './selectors'

import { actions } from 'state/books'
import Category from 'molecules/category'

import type { Category as CategoryType } from 'types'
type Props = {
  // eslint-disable-next-line react/no-unused-prop-types
  match: { params: { year?: string } },
  categories: Array<CategoryType>,
  fetchBooks: (string) => void
}

class BookList extends React.Component {
  static defaultProps: Props
  props: Props

  static getYear (props) {
    return R.path(['match', 'params', 'year'], props)
  }

  componentDidMount () {
    const year = BookList.getYear(this.props)

    if (year) {
      this.props.fetchBooks(year)
    }
  }

  componentWillUpdate (nextProps: Props) {
    const year = BookList.getYear(this.props)
    const newYear = BookList.getYear(nextProps)

    if (year && newYear && year !== newYear) {
      this.props.fetchBooks(newYear)
    }
  }

  render () {
    const { categories } = this.props
    return (
      <div>
        {categories.map(({name, books}) =>
          <Category key={name} name={name} books={books} />
        )}
      </div>
    )
  }
}

BookList.defaultProps = {
  match: { params: {} },
  categories: [],
  fetchBooks: (_) => undefined
}

export default connect(
  (state, props) =>
    createStructuredSelector({
      categories: booksSelector
    })(state),
  R.pick(['fetchBooks'])(actions)
)(BookList)
