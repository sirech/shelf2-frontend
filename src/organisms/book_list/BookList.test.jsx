import React from 'react'
import renderer from 'react-test-renderer'

import { fullRender } from '../../test'
import state from '../../state/__fixtures__'
import BookList, { BookList as TestBookList } from './BookList'

describe('components', () => {
  describe('BookList', () => {
    it('renders correctly', () => {
      const { component } = fullRender(<BookList />, state())
      expect(component.toJSON()).toMatchSnapshot()
    })

    it('fetches the books when there is a change in the year', () => {
      const match = { params: { year: '2017' } }
      const fetchBooks = jest.fn()

      renderer.create(<TestBookList match={match} fetchBooks={fetchBooks} />)
      expect(fetchBooks.mock.calls.length).toBe(1)
      expect(fetchBooks.mock.calls[0][0]).toBe('2017')
    })
  })
})
