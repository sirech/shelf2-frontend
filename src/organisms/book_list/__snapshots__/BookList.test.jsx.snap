// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components BookList renders correctly 1`] = `
<div>
  <div
    className="mb-3 card"
  >
    <div
      className="card-header"
    >
      <h4
        className="d-inline-block"
      >
        Software
      </h4>
    </div>
    <ul
      className="list-unstyled mb-0"
    >
      <li
        className="book-list-item"
      >
        <div
          className="book"
        >
          <h5>
            Elegant Objects
          </h5>
          <span
            className="stars"
          >
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m26.9 22.4l6.8-6.6-9.4-1.4-4.2-8.5-4.2 8.5-9.5 1.4 6.9 6.6-1.7 9.4 8.5-4.4 8.4 4.4z m11.7-8q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 1.1-0.9 1.1-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m26.9 22.4l6.8-6.6-9.4-1.4-4.2-8.5-4.2 8.5-9.5 1.4 6.9 6.6-1.7 9.4 8.5-4.4 8.4 4.4z m11.7-8q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 1.1-0.9 1.1-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
          </span>
        </div>
      </li>
      <li
        className="book-list-item"
      >
        <div
          className="book"
        >
          <h5>
            Site reliability engineering
          </h5>
          <span
            className="stars"
          >
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m26.9 22.4l6.8-6.6-9.4-1.4-4.2-8.5-4.2 8.5-9.5 1.4 6.9 6.6-1.7 9.4 8.5-4.4 8.4 4.4z m11.7-8q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 1.1-0.9 1.1-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
          </span>
        </div>
      </li>
      <li
        className="book-list-item"
      >
        <div
          className="book"
        >
          <h5>
            The hard thing about hard things
          </h5>
          <span
            className="stars"
          >
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m26.9 22.4l6.8-6.6-9.4-1.4-4.2-8.5-4.2 8.5-9.5 1.4 6.9 6.6-1.7 9.4 8.5-4.4 8.4 4.4z m11.7-8q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 1.1-0.9 1.1-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="mb-3 card"
  >
    <div
      className="card-header"
    >
      <h4
        className="d-inline-block"
      >
        History
      </h4>
    </div>
    <ul
      className="list-unstyled mb-0"
    >
      <li
        className="book-list-item"
      >
        <div
          className="book"
        >
          <h5>
            Genghis Khan and the making of the modern world
          </h5>
          <span
            className="stars"
          >
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m38.6 14.4q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 0.5-0.2 0.8t-0.7 0.3q-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m26.9 22.4l6.8-6.6-9.4-1.4-4.2-8.5-4.2 8.5-9.5 1.4 6.9 6.6-1.7 9.4 8.5-4.4 8.4 4.4z m11.7-8q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 1.1-0.9 1.1-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="star"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1.5em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1.5em"
              >
                <g>
                  <path
                    d="m26.9 22.4l6.8-6.6-9.4-1.4-4.2-8.5-4.2 8.5-9.5 1.4 6.9 6.6-1.7 9.4 8.5-4.4 8.4 4.4z m11.7-8q0 0.5-0.5 1.1l-8.1 7.9 1.9 11.2q0 0.1 0 0.4 0 1.1-0.9 1.1-0.4 0-0.9-0.2l-10-5.3-10 5.3q-0.5 0.2-0.9 0.2-0.5 0-0.7-0.3t-0.3-0.8q0-0.1 0.1-0.4l1.9-11.2-8.1-7.9q-0.6-0.6-0.6-1.1 0-0.8 1.3-1l11.2-1.6 5-10.2q0.4-0.9 1.1-0.9t1.1 0.9l5 10.2 11.2 1.6q1.2 0.2 1.2 1z"
                  />
                </g>
              </svg>
            </span>
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
